src/errors.rs

```diff
@@ -24,8 +24,8 @@ pub enum Error {
     #[error("failed to parse uri")]
     UriParse,
 
-    #[error("file is missing")]
-    FileMissing,
+    #[error("file '{0}' is missing")]
+    FileMissing(String),
```

Comment:
I would use `FileMissing(Box<Path>)` here:

- I am not a fan of using `String` for filenames
- `Box<>` saves somes bytes

Reply:
Implemented the changes, but haven't tested them yet